### ISSUE TITLE
change to diagnose_design and compare_designs

### ISSUE THREAD

**Issue by graemeblair** [DeclareDesign/DeclareDesign#128]
(1) diagnose_design will now take either a single design object or a list of them, with the same syntax

diagnose_design(design = list(design1, design2)) or diagnose_design(design = design1)

It will return the same diagnosis output, except that there will be an addition column if there is more than one design, design_label, which will be a number if there is no label in the design.

(2) compare_designs will keep its functionality to vary parts of the design. Possibly restricted to accepting a single design object rather than a list (but need to think about what would be best for comparison). The point is to make clear diagnose_design is for diagnosing one or more complete designs, and compare is to think about variations. Here's how it might work:

But for example when I compare two designs I might be interested in whether A really has more power than B? How would I do this? Presumably I want the SAME potential outcomes and population but possibly different sampling and estimation, then using the SAME simulations, draw a population, calculate ps in each design for the same data (though possibly different samples) and do the bootstrapping based on the common simulations (rather than two independent bootstraps). Output statistics would be things like how much more or less biased one design is relative to another and how confidencet we are in the difference....

**Comment by gerasy1987** [comment]
One comment on the new diagnose_design(): right now if we run

```
diagnosis <- diagnose_design(design = design1)
diagnosis$simulations
```

it will return simulations data frame for the diagnosis of design1. This is relevant for the current version of plot method for coverage since it uses the simulations data frame. So we need to think of a way to incorporate the new diagnose_design() functionality into plotting methods. For example we can probably do smth like

```
diagnosis <- diagnose_design(design = list(design1, design2))
plot(diagnosis, design = design1)
```

**Comment by jaspercooper** [comment]
Good point. If you look at how compare_designs() currently handles lists of
designs, you can see that there is a "diagnosis" index that gets added to
the simulations and diagnosands df (probably "design" would have been a
better label in hindsight). This is the exact same functionality that we
are going to transfer over to diagnose_design() by default.

So your suggestion would be relatively easy to implement, it would just
involve subsetting according to the index, i.e. plot(diagnosis, design = 1)
or plot(diagnosis, design = c(1,2,3)).

**Comment by gerasy1987** [comment]
Thanks Jasper, then it will be fairly easy.

One request (_if this is possible_) for the implementation of the new functionality of `diagnose_design()`. For both `summary.diagnosis` and `plot.diagnosis`, it will be really useful, if the additional column in the diagnosis object in case of a list of designs (e.g. `diagnose_design(design = list(design1, design2))`), would be called "daignosis_label" or "design_label", rather then "diagnosis" (as in current version of `compare_designs()`) or "design". This will allow to use unified approach to subsetting the diagnosis object based on estimand, estimator, estimate or design label.

**Comment by jaspercooper** [comment]
That makes a lot of sense. Will implement those changes to compare_designs
already.
